### ISSUE TITLE
@updatedAt is not allowed on "type" schema

### ISSUE THREAD

**Issue by kaminskypavel** [prisma/prisma#16629]
### Bug description

```
type Dates {
  createdAt DateTime? @default(now())
  updatedAt DateTime? @updatedAt
  viewedAt  DateTime? 
  deletedAt DateTime?
}
```

gives 

```
error: Attribute not known: "@updatedAt".
  -->  schema.prisma:194
   | 
193 |   createdAt DateTime? @default(now())
194 |   updatedAt DateTime? @updatedAt
   | ```

though the docs are not permitting that 

https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#updatedat



### How to reproduce

read Bug description


### Expected behavior

_No response_

### Prisma information

type Dates {
  createdAt DateTime? @default(now())
  updatedAt DateTime? @updatedAt
  viewedAt  DateTime? 
  deletedAt DateTime?
}


### Environment & setup

- OS: win11
- Database: MongoDb
- Node.js version: node18


### Prisma Version

```
prisma                  : 4.7.1
@prisma/client          : 4.7.1
Current platform        : windows
```

**Comment by janpio** [comment]
This is currently not supported. We should update the documentation for now, and then see what the use cases for it would be.

**Comment by kaminskypavel** [comment]
I honor your decision but the use case here is pretty straightforward. 
consider 

``
model User {
   name String
   dates? Dates
}


type Dates {
  createdAt DateTime? @default(now())
  updatedAt DateTime? @updatedAt
  viewedAt  DateTime? 
  deletedAt DateTime?
}
```

how should I tackle this?

**Comment by janpio** [comment]
You can't right now, unless you hand write the update of the `updatedAt` value. This issue now tracks this feature request.

**Comment by Amrbahaa22** [comment]
Any news on when this feature will be available?

**Comment by janpio** [comment]
No, this has not been prioritized. This issue is used to collect demand and additional use cases which might lead to use deciding to implement this.

**Comment by zthart** [comment]
> This issue is used to collect demand and additional use cases

Not supporting this could cause mongodb users to follow some antipatterns if they want to take advantage of the feature. There's a "accessed together, stored together" paradigm that's pushed pretty heavily since the alternative is just issuing more queries - either making additional `.find` queries or needing some `$lookup` stage in an aggregation pipeline.

Take this basic implementation for a simple day-tracker application that allows users to create daily planning entries that include a todo-list per day. I don't want todos to roll over into the next day, so it would actually be pretty reasonable to store todos as a subdocument array on the individual day entry documents, especially if you always need both the day entry data and the todolist data together.

```
model User {
  name String
  email String @unique
}

model DayEntry {
  userId String @db.ObjectId
  summary String
  todos Todo[]
  createdAt DateTime @default(now())
}

type Todo {
  uuid String @default(uuid())
  title String
  isComplete boolean
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  lastCompletedAt DateTime?
}
```

It would be a very different design to need to create a separate Todo model and use relations in this case to get the updatedAt support, and could end up being much less efficient in terms of db load. i.e. one query to get a set of entries and their corresponding todos vs. >1 if they were a relation that needed to be queried.

There are definitely real-world cases where subdocuments or subdocument arrays could benefit from use of the `@updatedAt` feature beyond sharing a `Dates` type that you can use across your models (although this would be nice to prevent duplication in the prisma schema if many of your models will need these kinds of timestamps). The ability to combine the "normalized" relationships like between User and DayEntry with embedded relationships like the Todo array here where each is appropriate is a strength of MongoDB.

**Comment by Nik-Novak** [comment]
I agree, I think that to properly support mongodb patterns this is important
